src/components/Map.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { useEffect, useRef } from 'react';
+
+export const Map = ({
+  x,
+  y,
+  zoneId,
+  scenarioId,
+}: {
+  x?: number;
+  y?: number;
+  zoneId?: number;
+  scenarioId?: number;
+}): JSX.Element => {
+  const canvasElement = useRef<HTMLCanvasElement>(null);
+
+  useEffect(() => {
+    if (canvasElement.current) {
+      const canvasWidth = canvasElement.current.width;
+      canvasElement.current.height = canvasWidth;
+      const xTranslated = x ? (x / 65535) * canvasWidth : undefined;
+      const yTranslated = y ? (y / 65535) * canvasWidth : undefined;
+      const ctx = canvasElement.current.getContext('2d');
+      if (ctx) {
+        const image = new Image();
+        const skullImage = new Image();
+        image.src = scenarioId
```

Comment:
this needs to always use zoneId. Scenario maps are also using zoneid.

Some scenarios like ranked reuse same zone id for 2-3 different scenarios

Reply:
Not sure I understand the second part of your comment correctly but changed it in the last commit.

Reply:
> Not sure I understand the second part of your comment correctly but changed it in the last commit.

multiple scenarios can share the same zoneId.

For example all these 4 scenarios have same zoneId:

```
  2104: 'Maw of Madness',
  2152: 'Maw of Madness (Ranked Solo)',
  2153: 'Maw of Madness (Ranked)',
  2158: 'Daemonball Rally',
```